pkg/kv/kvserver/replica_command.go

```diff
@@ -1289,8 +1289,15 @@ func (r *Replica) maybeLeaveAtomicChangeReplicasAndRemoveLearners(
 		return nil, err
 	}
 
+	// If we detect that there are learners in the process of receiving their
+	// initial upreplication snapshot, we don't want to remove them and we bail
+	// out.
+	if r.hasOutstandingLearnerSnapshotInFlight() {
+		return desc, errors.New("cannot remove learner while a snapshot is in flight")
```

Comment:
Nothing wrong with this code, but it'll give us new test flakes.

Reply:
I've made it such that we don't wait out the grace period for snapshot application when this method checks for the presence of an in-flight snapshot. All the flakes I was seeing (thus far) were because of that 3-second grace period.